staging/src/k8s.io/client-go/tools/clientcmd/validation.go

```diff
@@ -235,10 +234,15 @@ func validateAuthInfo(authInfoName string, authInfo clientcmdapi.AuthInfo) []err
 				validationErrors = append(validationErrors, fmt.Errorf("unable to read client-key %v for %v due to %v", authInfo.ClientKey, authInfoName, err))
 			}
 		}
+		methods = append(methods, "tlsClientAuth")
+	}
+
+	if authInfo.AuthProvider != nil {
+		methods = append(methods, "authProviderAuth")
 	}
 
 	// authPath also provides information for the client to identify the server, so allow multiple auth methods in that case
-	if (len(methods) > 1) && (!usingAuthPath) {
+	if len(methods) > 1 {
```

Comment:
For some reason, I thought some of the auth methods would actually persist a bearer token back into the config file. is that not the case?

Reply:
It is the case. I don't follow how that relates.

Reply:
I mean in the top level bearerToken field, where it would be counted as an auth method, and trigger an error if bearer token and auth provider both counted as auth methods